File: src/ros2_bridge/mappers/geometry_msgs/point32.rs
//! Typed mapper for `geometry_msgs/msg/Point32`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn point32_to_bus(msg: ros_env::geometry_msgs::msg::Point32) -> crate::geometry_msgs::msg::v1::Point32 {
    crate::geometry_msgs::msg::v1::Point32 {
        x: msg.x,
        y: msg.y,
        z: msg.z,
    }
}

pub(crate) fn point32_to_ros(bus: crate::geometry_msgs::msg::v1::Point32) -> ros_env::geometry_msgs::msg::Point32 {
    ros_env::geometry_msgs::msg::Point32 {
        x: bus.x,
        y: bus.y,
        z: bus.z,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsPoint32Mapper;

impl TypedTopicMapper for GeometryMsgsPoint32Mapper {
    type Ros = ros_env::geometry_msgs::msg::Point32;
    type Bus = crate::geometry_msgs::msg::v1::Point32;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(point32_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(point32_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/point32.py
"""Generated mapper for `geometry_msgs/msg/Point32`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.geometry_msgs.msg.v1 import Point32 as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def point32_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.x = msg.x
    bus.y = msg.y
    bus.z = msg.z
    return bus


def point32_to_ros(bus):
    from geometry_msgs.msg import Point32 as RosMsg

    out = RosMsg()
    out.x = bus.x
    out.y = bus.y
    out.z = bus.z
    return out


class GeometryMsgsPoint32Mapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from geometry_msgs.msg import Point32 as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return point32_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return point32_to_ros(bus)
